### DISCUSSION TITLE
3.0 Roadmap

### DISCUSSION THREAD

**Discussion by jankapunkt** [TheRealNate/react-native-meteor#137] [Ideas]
Meteor 3.0 is closing in to Beta level and there will be huge changes. Some of them may affect this library, too and we might need major/breaking changes, too, in order to stay with the ecosystem.

I would also like to this chance to improve this library with, maybe even braking chances, since Meteor 3.0 will be breaking for all of us, anyway.

Topics I think about, among others:

- rewrite to ES6 or, better, TypeScript
- offline-first (see other discussions, #104, #127)
- async Tracker (which is already part of Meteor 2.x)
- React-suspense hooks (which is also already part of Meteor 2.x)
- improve test coverage and add E2E tests (we can use repos like the Mattermost app or BlueSky app to see how they do it)
- what else?

**Comment by not-ai-yet** [comment]
Great conversation to kickstart!
- Rewrite to TypeScript would be fantastic IMO
- For offline first, I'm thinking about if it would make sense to bring that into the core package or have a separate enhancement package to make the core less opinionated

Agree on all other points mentioned

**Reply by jankapunkt** [reply]
I would also favor a companion packages for offline-first but they should be pluggable without the need for monkey-patching or similar.

**Comment by jankapunkt** [comment]
We need better logging of internals, which is crucial in production to avoid long debugging session with issues like #141 

A potential solution should make it available to catch any potential error and log multiple levels of internals.
